src/main/java/dev/sigstore/json/GsonSupplier.java

```diff
@@ -27,14 +27,19 @@
  * requests between sigstore and this client -- and should probably be used for any api call to
  * sigstore that expects JSON.
  */
-public class GsonSupplier implements Supplier<Gson> {
+public enum GsonSupplier implements Supplier<Gson> {
```

Comment:
I don't know if https://immutables.github.io/immutable.html#singleton-instances matters, but they have singleton support.

Reply:
GTK. I'm not sure about either pattern right now, so hopefully I'll have more of an opinion once I get back into the swing of things.